pkgs/desktops/plasma-5/plasma-workspace/default.nix

```diff
@@ -59,6 +59,13 @@ mkDerivation {
       --replace 'query_qmake(QtBinariesDir QT_INSTALL_BINS)' 'set(QtBinariesDir "${lib.getBin qttools}/bin")'
   '';
 
+  postFixup = ''
```

Comment:
Can't we add argv0 to makeWrapper?

Also this needs a comment.

Reply:
the exec already has -a but for some reason (atleast) wayland programs dont honor it always 
```
exec -a "$0" "/nix/store/xpabq70sg4sjpzxz0755sbrmqrwq2q6z-plasma-workspace-5.22.4/bin/.plasmashell-wrapped"  "$@"
```

Reply:
I think they're using `/proc/self/exe` (or similar) instead of `argv[0]`

Reply:
> I think they're using /proc/self/exe (or similar) instead of argv[0]

related https://github.com/NixOS/nixpkgs/pull/116549

Reply:
moving wrapped binaries to $out/.bin (which im experimenting with) would prevent us from needing to maintain a lot of patches

Reply:
This will break programs, which refers to executables in its own `/bin`, as stated here: https://github.com/NixOS/nixpkgs/commit/9deb7f8aae431ed7725cfaa13edf8645d19d91f2#diff-b8b59bce5985691f69d9ba73cc1ac4125f023ca1e6d8f5364fedffabf64fb3a2

Reply:
Moving wrapped programs is going to be a huge uphill battle against the current status quo. While I agree it would be the more correct solution, the pragmatic solution, I think, is to maintain some patches as close to the root of the problem as possible, while in parallel the wrappers can be re-thought entirely.